templates/routes-for-calico.yaml

```diff
@@ -1,89 +1,82 @@
-kind: Deployment
+apiVersion: apps/v1beta2
+kind: DaemonSet
 metadata:
   name: {{ template "CGW.fullname" . }}-calico-setup
   namespace: kube-system
-  label: 
+  labels: 
     app: {{ template "CGW.name" . }}
     chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "CGW.name" . }}
+      release: {{ .Release.Name }}
   template:
     metadata:
       labels:
         app: {{ template "CGW.name" . }}
         release: {{ .Release.Name }}
     spec:
-      serviceAccountName: {{ template "CGW.fullname" .}}-calico-setup
-      initContainers:
-      - name: {{ .Chart.Name }}-calico-etcd-routing-enabler
-        image: "quay.io/coreos/etcd:latest"
-        command:
-          - /bin/sh
-          - "-xc"
-          - >
-              etcdctl --endpoint=$ENDPOINT set
-              calico/bgp/v1/global/custom_filters/v4/ipsec
-              "if ( net = $REMOTENET ) then { accept ; }"
-        env:
-        - name: ENDPOINT
-          valueFrom:
-            configMapKeyRef:
-              name: calico-config
-              key: etcd_endpoints
-        - name: REMOTENET
-          value: "{{ .Values.ipsec.remotenet }}"
+      hostNetwork: true
+      {{- if .Values.nodeSelector }}
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: "{{ .Values.calicoSetup.excludeHost.key }}"
+                operator: NotIn
+                values:
+                - "{{ .Values.calicoSetup.excludeHost.value }}"
+      {{- end }}
       containers:
-      - name: {{ .Chart.Name }}-calico-setup
-        image: "{{ .Values.calicoSetup.image.repository }}:{{ .Values.calicoSetup.image.tag }}"
+      - name: {{ .Chart.Name }}-calico-setup 
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        image: alpine:latest
+        command:
+        - sh
+        - -c
+        - |
+          set -euo pipefail
+
+          _remove_route() {
+              echo "ip route del $IPSEC_REMOTENET via $TUNNEL_GATEWAY dev $TUNNEL_INTERFACE onlink"
+              ip route del $IPSEC_REMOTENET via $TUNNEL_GATEWAY dev $TUNNEL_INTERFACE onlink
+              return 0
+          }
+
+          _add_route() {
+              echo "======= setup route ======="
+              echo "ip route add $IPSEC_REMOTENET via $TUNNEL_GATEWAY dev $TUNNEL_INTERFACE onlink"
+              ip route add $IPSEC_REMOTENET via $TUNNEL_GATEWAY dev $TUNNEL_INTERFACE onlink
+          }
+
+          _term() {
+              echo "======= caught SIGTERM signal ======="
+              _remove_route
+              exit 0
+          }
+
+          _add_route
+          trap _term TERM INT
+          while true
+          do
+            echo "going to sleep for 1d"
+            sleep 1d
```

Comment:
sleeping for one day will delay handling traps for one day. this renders your trap almost useless.